tests/Feature/Api/V1/GameListTest.php

```diff
@@ -165,6 +165,25 @@ public function testGetGameList(): void
                 ],
             ]);
 
+        // games for system 1 with offset and count
+        /* disabled due to an error with SQL_CALC_FOUND_ROWS in testing */
+        // $this->get($this->apiUrl('GetGameList', ['i' => $system1->ID, 'o' => 1, 'c' => 1]))
+        //     ->assertSuccessful()
+        //     ->assertJsonCount(1)
+        //     ->assertJson([
+        //         [
+        //             'ID' => $game2->ID,
+        //             'Title' => $game2->Title,
+        //             'ConsoleID' => $system1->ID,
+        //             'ConsoleName' => $system1->Name,
+        //             'ImageIcon' => $game2->ImageIcon,
+        //             'NumAchievements' => 0,
+        //             'Points' => 0,
+        //             'NumLeaderboards' => 0,
+        //             'ForumTopicID' => null,
+        //         ],
+        //     ]);
+
```

Comment:
We'll need to find a way to get this enabled.

You'll likely need to add a compatibility layer to the query users _app/Helpers/util/database.php_.

Reply:
Any idea about how to do it? I'm an iOS dev and PHP is not really fresh in my mind 😅

Reply:
> Any idea about how to do it? I'm an iOS dev and PHP is not really fresh in my mind 😅

I swear this isn't my alt

Reply:
@WedgeSparda Sorry, other than suggesting where & how to remediate the issue, my capacity to help is stretched really thin right now.

Reply:
@wescopeland 
I'm digging into and found a possible reason of the problem. When running the code outside of the test, is working, but in the test isn't since since the new params are not reaching the endpoint.

What I mean is that in my test I have this
```
$this->get($this->apiUrl('GetGameList', ['i' => $system1->ID, 'o' => 1, 'c' => 1]))
```
But when printing the values in the endpoint, both are 0 and the test fails due to an error. 

Here's the complete trace. Is like the two new params are not reaching the endpoint from the test.

```
PHPUnit 10.5.10 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.7
Configuration: /RAWeb/phpunit.xml

.F                                                                  2 / 2 (100%)

Time: 00:01.458, Memory: 70.50 MB

There was 1 failure:

1) Tests\Feature\Api\V1\GameListTest::testGetGameList
Expected response status code [>=200, <300] but received 500.
Failed asserting that false is true.

The following exception occurred during the last request:

PDOException: SQLSTATE[HY000]: General error: 1 near ".": syntax error in /RAWeb/vendor/laravel/framework/src/Illuminate/Database/Connection.php:423
Stack trace:
#0 /RAWeb/vendor/laravel/framework/src/Illuminate/Database/Connection.php(423): PDO->prepare('SELECT SQL_CALC...')
#1 /RAWeb/vendor/laravel/framework/src/Illuminate/Database/Connection.php(816): Illuminate\Database\Connection->Illuminate\Database\{closure}('SELECT SQL_CALC...', Array)
#2 /RAWeb/vendor/laravel/framework/src/Illuminate/Database/Connection.php(783): Illuminate\Database\Connection->runQueryCallback('SELECT SQL_CALC...', Array, Object(Closure))
#3 /RAWeb/vendor/laravel/framework/src/Illuminate/Database/Connection.php(414): Illuminate\Database\Connection->run('SELECT SQL_CALC...', Array, Object(Closure))
#4 /RAWeb/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php(509): Illuminate\Database\Connection->select('SELECT SQL_CALC...', Array)
#5 /RAWeb/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(355): Illuminate\Database\DatabaseManager->__call('select', Array)
#6 /RAWeb/app/Helpers/util/database.php(39): Illuminate\Support\Facades\Facade::__callStatic('select', Array)
#7 /RAWeb/app/Helpers/util/database.php(16): legacyDbSelect('SELECT SQL_CALC...', Array)
#8 /RAWeb/app/Helpers/database/game.php(403): legacyDbFetchAll('SELECT SQL_CALC...', Array)
#9 /RAWeb/public/API/API_GetGameList.php(40): getGamesListByDev(NULL, 1, Array, 1, false, 2, 1, 1)
#10 /RAWeb/app/Http/Concerns/HandlesPublicFileRequests.php(46): require('/Users/roberto....')
#11 /RAWeb/app/Api/RouteServiceProvider.php(90): App\Api\RouteServiceProvider->handleRequest('API/API_GetGame...')
#12 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/CallableDispatcher.php(40): App\Api\RouteServiceProvider->App\Api\{closure}('API_GetGameList')
#13 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Route.php(237): Illuminate\Routing\CallableDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Closure))
#14 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Route.php(208): Illuminate\Routing\Route->runCallable()
#15 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Router.php(806): Illuminate\Routing\Route->run()
#16 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#17 /RAWeb/app/Api/Middleware/LogApiUsage.php(20): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#18 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): App\Api\Middleware\LogApiUsage->handle(Object(Illuminate\Http\Request), Object(Closure))
#19 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#20 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure))
#21 /RAWeb/vendor/laravel/framework/src/Illuminate/Auth/Middleware/Authenticate.php(57): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#22 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Auth\Middleware\Authenticate->handle(Object(Illuminate\Http\Request), Object(Closure), 'api-token')
#23 /RAWeb/app/Http/Middleware/JsonResponse.php(16): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#24 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): App\Http\Middleware\JsonResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#25 /RAWeb/app/Api/Middleware/AccessControlAllowOriginWildcard.php(14): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#26 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): App\Api\Middleware\AccessControlAllowOriginWildcard->handle(Object(Illuminate\Http\Request), Object(Closure))
#27 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#28 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Router.php(805): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#29 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Router.php(784): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#30 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Router.php(748): Illuminate\Routing\Router->runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))
#31 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Router.php(737): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#32 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(200): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#33 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#34 /RAWeb/vendor/livewire/livewire/src/Features/SupportDisablingBackButtonCache/DisableBackButtonCacheMiddleware.php(19): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#35 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Livewire\Features\SupportDisablingBackButtonCache\DisableBackButtonCacheMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#36 /RAWeb/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php(59): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#37 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Barryvdh\Debugbar\Middleware\InjectDebugbar->handle(Object(Illuminate\Http\Request), Object(Closure))
#38 /RAWeb/app/Http/Middleware/FeatureFlagMiddleware.php(26): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#39 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): App\Http\Middleware\FeatureFlagMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#40 /RAWeb/vendor/spatie/laravel-robots-middleware/src/RobotsMiddleware.php(14): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#41 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Spatie\RobotsMiddleware\RobotsMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#42 /RAWeb/vendor/spatie/laravel-missing-page-redirector/src/RedirectsMissingPages.php(12): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#43 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Spatie\MissingPageRedirector\RedirectsMissingPages->handle(Object(Illuminate\Http\Request), Object(Closure))
#44 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#45 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#46 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull->handle(Object(Illuminate\Http\Request), Object(Closure))
#47 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#48 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#49 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\TrimStrings->handle(Object(Illuminate\Http\Request), Object(Closure))
#50 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#51 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle(Object(Illuminate\Http\Request), Object(Closure))
#52 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(99): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#53 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle(Object(Illuminate\Http\Request), Object(Closure))
#54 /RAWeb/vendor/laravel/framework/src/Illuminate/Http/Middleware/HandleCors.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#55 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Http\Middleware\HandleCors->handle(Object(Illuminate\Http\Request), Object(Closure))
#56 /RAWeb/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#57 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Http\Middleware\TrustProxies->handle(Object(Illuminate\Http\Request), Object(Closure))
#58 /RAWeb/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustHosts.php(48): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#59 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Http\Middleware\TrustHosts->handle(Object(Illuminate\Http\Request), Object(Closure))
#60 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#61 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(175): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#62 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(144): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#63 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php(585): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#64 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php(346): Illuminate\Foundation\Testing\TestCase->call('GET', 'API/API_GetGame...', Array, Array, Array, Array)
#65 /RAWeb/tests/Feature/Api/V1/GameListTest.php(170): Illuminate\Foundation\Testing\TestCase->get('API/API_GetGame...')
#66 /RAWeb/vendor/phpunit/phpunit/src/Framework/TestCase.php(1122): Tests\Feature\Api\V1\GameListTest->testGetGameList()
#67 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Testing/TestCase.php(61): PHPUnit\Framework\TestCase->runTest()
#68 /RAWeb/vendor/phpunit/phpunit/src/Framework/TestCase.php(654): Illuminate\Foundation\Testing\TestCase->runTest()
#69 /RAWeb/vendor/phpunit/phpunit/src/Framework/TestRunner.php(104): PHPUnit\Framework\TestCase->runBare()
#70 /RAWeb/vendor/phpunit/phpunit/src/Framework/TestCase.php(488): PHPUnit\Framework\TestRunner->run(Object(Tests\Feature\Api\V1\GameListTest))
#71 /RAWeb/vendor/phpunit/phpunit/src/Framework/TestSuite.php(340): PHPUnit\Framework\TestCase->run()
#72 /RAWeb/vendor/phpunit/phpunit/src/TextUI/TestRunner.php(63): PHPUnit\Framework\TestSuite->run()
#73 /RAWeb/vendor/phpunit/phpunit/src/TextUI/Application.php(198): PHPUnit\TextUI\TestRunner->run(Object(PHPUnit\TextUI\Configuration\Configuration), Object(PHPUnit\Runner\ResultCache\DefaultResultCache), Object(PHPUnit\Framework\TestSuite))
#74 /RAWeb/vendor/phpunit/phpunit/phpunit(104): PHPUnit\TextUI\Application->run(Array)
#75 /RAWeb/vendor/bin/phpunit(122): include('/Users/roberto....')
#76 {main}

Next Illuminate\Database\QueryException: SQLSTATE[HY000]: General error: 1 near ".": syntax error (Connection: sqlite, SQL: SELECT SQL_CALC_FOUND_ROWS gd.ID, gd.Title, gd.ImageIcon, gd.TotalTruePoints,
                     COALESCE(gd.achievements_published,0) AS NumAchievements,
                     gd.points_total AS MaxPointsAvailable,
                     CASE WHEN LENGTH(gd.RichPresencePatch) > 0 THEN 1 ELSE 0 END AS RichPresence,
                     CASE WHEN gd.points_total > 0 THEN ROUND(gd.TotalTruePoints/gd.points_total, 2) ELSE 0.00 END AS RetroRatio,
                     gd.ForumTopicID, gd.ConsoleID, c.Name as ConsoleName
                  FROM GameData gd
                  INNER JOIN Console c ON c.ID = gd.ConsoleID 
                  WHERE 1=1 AND gd.ConsoleID = 1
                  GROUP BY gd.ID ORDER BY IIF(gd.Title LIKE '~%', 1, 0), gd.Title LIMIT 1, 1) in /RAWeb/vendor/laravel/framework/src/Illuminate/Database/Connection.php:829
Stack trace:
#0 /RAWeb/vendor/laravel/framework/src/Illuminate/Database/Connection.php(783): Illuminate\Database\Connection->runQueryCallback('SELECT SQL_CALC...', Array, Object(Closure))
#1 /RAWeb/vendor/laravel/framework/src/Illuminate/Database/Connection.php(414): Illuminate\Database\Connection->run('SELECT SQL_CALC...', Array, Object(Closure))
#2 /RAWeb/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php(509): Illuminate\Database\Connection->select('SELECT SQL_CALC...', Array)
#3 /RAWeb/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(355): Illuminate\Database\DatabaseManager->__call('select', Array)
#4 /RAWeb/app/Helpers/util/database.php(39): Illuminate\Support\Facades\Facade::__callStatic('select', Array)
#5 /RAWeb/app/Helpers/util/database.php(16): legacyDbSelect('SELECT SQL_CALC...', Array)
#6 /RAWeb/app/Helpers/database/game.php(403): legacyDbFetchAll('SELECT SQL_CALC...', Array)
#7 /RAWeb/public/API/API_GetGameList.php(40): getGamesListByDev(NULL, 1, Array, 1, false, 2, 1, 1)
#8 /RAWeb/app/Http/Concerns/HandlesPublicFileRequests.php(46): require('/Users/roberto....')
#9 /RAWeb/app/Api/RouteServiceProvider.php(90): App\Api\RouteServiceProvider->handleRequest('API/API_GetGame...')
#10 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/CallableDispatcher.php(40): App\Api\RouteServiceProvider->App\Api\{closure}('API_GetGameList')
#11 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Route.php(237): Illuminate\Routing\CallableDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Closure))
#12 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Route.php(208): Illuminate\Routing\Route->runCallable()
#13 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Router.php(806): Illuminate\Routing\Route->run()
#14 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#15 /RAWeb/app/Api/Middleware/LogApiUsage.php(20): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#16 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): App\Api\Middleware\LogApiUsage->handle(Object(Illuminate\Http\Request), Object(Closure))
#17 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#18 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure))
#19 /RAWeb/vendor/laravel/framework/src/Illuminate/Auth/Middleware/Authenticate.php(57): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#20 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Auth\Middleware\Authenticate->handle(Object(Illuminate\Http\Request), Object(Closure), 'api-token')
#21 /RAWeb/app/Http/Middleware/JsonResponse.php(16): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#22 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): App\Http\Middleware\JsonResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#23 /RAWeb/app/Api/Middleware/AccessControlAllowOriginWildcard.php(14): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#24 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): App\Api\Middleware\AccessControlAllowOriginWildcard->handle(Object(Illuminate\Http\Request), Object(Closure))
#25 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#26 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Router.php(805): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#27 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Router.php(784): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#28 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Router.php(748): Illuminate\Routing\Router->runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))
#29 /RAWeb/vendor/laravel/framework/src/Illuminate/Routing/Router.php(737): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#30 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(200): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#31 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#32 /RAWeb/vendor/livewire/livewire/src/Features/SupportDisablingBackButtonCache/DisableBackButtonCacheMiddleware.php(19): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#33 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Livewire\Features\SupportDisablingBackButtonCache\DisableBackButtonCacheMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#34 /RAWeb/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php(59): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#35 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Barryvdh\Debugbar\Middleware\InjectDebugbar->handle(Object(Illuminate\Http\Request), Object(Closure))
#36 /RAWeb/app/Http/Middleware/FeatureFlagMiddleware.php(26): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#37 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): App\Http\Middleware\FeatureFlagMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#38 /RAWeb/vendor/spatie/laravel-robots-middleware/src/RobotsMiddleware.php(14): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#39 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Spatie\RobotsMiddleware\RobotsMiddleware->handle(Object(Illuminate\Http\Request), Object(Closure))
#40 /RAWeb/vendor/spatie/laravel-missing-page-redirector/src/RedirectsMissingPages.php(12): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#41 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Spatie\MissingPageRedirector\RedirectsMissingPages->handle(Object(Illuminate\Http\Request), Object(Closure))
#42 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#43 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#44 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull->handle(Object(Illuminate\Http\Request), Object(Closure))
#45 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#46 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#47 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\TrimStrings->handle(Object(Illuminate\Http\Request), Object(Closure))
#48 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#49 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle(Object(Illuminate\Http\Request), Object(Closure))
#50 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(99): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#51 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle(Object(Illuminate\Http\Request), Object(Closure))
#52 /RAWeb/vendor/laravel/framework/src/Illuminate/Http/Middleware/HandleCors.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#53 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Http\Middleware\HandleCors->handle(Object(Illuminate\Http\Request), Object(Closure))
#54 /RAWeb/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#55 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Http\Middleware\TrustProxies->handle(Object(Illuminate\Http\Request), Object(Closure))
#56 /RAWeb/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustHosts.php(48): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#57 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(183): Illuminate\Http\Middleware\TrustHosts->handle(Object(Illuminate\Http\Request), Object(Closure))
#58 /RAWeb/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#59 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(175): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#60 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(144): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#61 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php(585): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#62 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php(346): Illuminate\Foundation\Testing\TestCase->call('GET', 'API/API_GetGame...', Array, Array, Array, Array)
#63 /RAWeb/tests/Feature/Api/V1/GameListTest.php(170): Illuminate\Foundation\Testing\TestCase->get('API/API_GetGame...')
#64 /RAWeb/vendor/phpunit/phpunit/src/Framework/TestCase.php(1122): Tests\Feature\Api\V1\GameListTest->testGetGameList()
#65 /RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Testing/TestCase.php(61): PHPUnit\Framework\TestCase->runTest()
#66 /RAWeb/vendor/phpunit/phpunit/src/Framework/TestCase.php(654): Illuminate\Foundation\Testing\TestCase->runTest()
#67 /RAWeb/vendor/phpunit/phpunit/src/Framework/TestRunner.php(104): PHPUnit\Framework\TestCase->runBare()
#68 /RAWeb/vendor/phpunit/phpunit/src/Framework/TestCase.php(488): PHPUnit\Framework\TestRunner->run(Object(Tests\Feature\Api\V1\GameListTest))
#69 /RAWeb/vendor/phpunit/phpunit/src/Framework/TestSuite.php(340): PHPUnit\Framework\TestCase->run()
#70 /RAWeb/vendor/phpunit/phpunit/src/TextUI/TestRunner.php(63): PHPUnit\Framework\TestSuite->run()
#71 /RAWeb/vendor/phpunit/phpunit/src/TextUI/Application.php(198): PHPUnit\TextUI\TestRunner->run(Object(PHPUnit\TextUI\Configuration\Configuration), Object(PHPUnit\Runner\ResultCache\DefaultResultCache), Object(PHPUnit\Framework\TestSuite))
#72 /RAWeb/vendor/phpunit/phpunit/phpunit(104): PHPUnit\TextUI\Application->run(Array)
#73 /RAWeb/vendor/bin/phpunit(122): include('/Users/roberto....')
#74 {main}

----------------------------------------------------------------------------------

SQLSTATE[HY000]: General error: 1 near ".": syntax error (Connection: sqlite, SQL: SELECT SQL_CALC_FOUND_ROWS gd.ID, gd.Title, gd.ImageIcon, gd.TotalTruePoints,
                     COALESCE(gd.achievements_published,0) AS NumAchievements,
                     gd.points_total AS MaxPointsAvailable,
                     CASE WHEN LENGTH(gd.RichPresencePatch) > 0 THEN 1 ELSE 0 END AS RichPresence,
                     CASE WHEN gd.points_total > 0 THEN ROUND(gd.TotalTruePoints/gd.points_total, 2) ELSE 0.00 END AS RetroRatio,
                     gd.ForumTopicID, gd.ConsoleID, c.Name as ConsoleName
                  FROM GameData gd
                  INNER JOIN Console c ON c.ID = gd.ConsoleID 
                  WHERE 1=1 AND gd.ConsoleID = 1
                  GROUP BY gd.ID ORDER BY IIF(gd.Title LIKE '~%', 1, 0), gd.Title LIMIT 1, 1)

/RAWeb/vendor/laravel/framework/src/Illuminate/Testing/TestResponse.php:92
/RAWeb/tests/Feature/Api/V1/GameListTest.php:171
/RAWeb/vendor/laravel/framework/src/Illuminate/Foundation/Testing/TestCase.php:61
```

Reply:
So, It seems `SQL_CALC_FOUND_ROWS` and `FOUND_ROWS` are [deprecated](https://dev.mysql.com/worklog/task/?id=12615). Not sure how can the SQL code can be updated to embrace a new way to do the same.

Reply:
An approach I would explore is trying to not even use `getGamesListByDev()`.

I can't help but notice this endpoint's response shape is not that complex. We should be able to achieve the same response shape using Eloquent ORM and bypass the helper altogether.

Again, beyond this observation, I don't have the capacity right now to help with implementation details.

Reply:
That sound like something I can try 💪